imap_processing/hi/l1c/hi_l1c.py

```diff
@@ -45,8 +50,8 @@ def hi_l1c(dependencies: list, data_version: str) -> xr.Dataset:
 
     # TODO: I am not sure what the input for Goodtimes will be so for now,
     #    If the input is an xarray Dataset, do pset processing
-    if len(dependencies) == 1 and isinstance(dependencies[0], xr.Dataset):
-        l1c_dataset = generate_pset_dataset(dependencies[0])
+    if len(dependencies) == 2 and isinstance(dependencies[0], xr.Dataset):
```

Comment:
Do we need to assert on the instance type of all dependencies, or are we sure these are going to be in the right order when we request them?